src/core/public/http/types.ts

```diff
@@ -52,4 +51,21 @@ export interface HttpFetchOptions extends HttpRequestInit {
   prependBasePath?: boolean;
   headers?: HttpHeadersInit;
 }
+export type HttpHandler = (path: string, options?: HttpFetchOptions) => Promise<HttpBody>;
+export interface IHttpService {
```

Comment:
We don't use the I-prefix convention in any of our types. So far we've been preferring the `HttpSetup` and `HttpStart` convention, even if they're the same type. I don't have a strong preference here as long as we're consistent.

If we stop exporting the types from `core/public` directly and instead only export `CoreSetup` and `CoreStart` this problem becomes less important.

As an aside, I think you could argue that we may not want any plugins to be making HTTP requests during the `setup` phase. We could leverage the `createPluginSetupContext` function to hide this functionality from plugins while keeping it available to Core.

Also this public type and it's properties should include ts docs.

Reply:
> As an aside, I think you could argue that we may not want any plugins to be making HTTP requests during the `setup` phase.

Is this statement accurate given capabilities' need to make requests at setup?